testnet/Dockerfile.debug

```diff
@@ -0,0 +1,46 @@
+# BUILD IMAGE ---------------------------------------------------------
+
+FROM rust:1.85.1-slim-bookworm AS builder
+
+# Install dependencies needed for building RocksDB and etcd.
+RUN apt-get update && apt-get install -yq \
+    git gcc g++ clang libssl-dev pkg-config ca-certificates curl cmake ninja-build
+
+COPY . /nomos
+
+WORKDIR /opt
+
+RUN git clone https://github.com/risc0/risc0.git && \
+    cd risc0 && \
+    cargo install --path rzup && \
+    rzup build rust --tag-or-commit 4d91de4e48198da2e33413efdcd9cd2cc0c46688 && \
+    git checkout origin/release-2.0 && \
+    cargo install --path risc0/cargo-risczero && \
+    rm -rf /usr/local/cargo/registry /usr/local/cargo/git && \
+    rm -rf /root/.risc0/tmp && \
+    rm -rf * && \
+    cd /nomos && \
+    cargo build && \
+    find . -type f ! -path "./target/debug/nomos-node" ! -path "./target/debug/nomos-executor" \
```

Comment:
Why is this step necessary? We anyway only copy the necessary binaries later on.

Reply:
This is probably also not required if we only compile the binaries we need, as mentioned in https://github.com/logos-co/nomos/pull/1212#discussion_r2031027709.

Reply:
It is what I believe make the layer smaller. I don't have precise knowledge about file per file requirements. Please note this few lines of code took 3 days to come up. Not that it cannot be further improved. Ideally we would be able to isolate risc0 to separate image. I'm not there yet.

Reply:
Similar approach used in Rust docker images. e.g.: https://github.com/rust-lang/docker-rust/blob/master/stable/bookworm/slim/Dockerfile